lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -8770,6 +8771,89 @@ using namespace Js;
             return scriptContext->GetLibrary()->CreateArrayIterator(thisObj, JavascriptArrayIteratorKind::Value));
     }
 
+    // Relative indexing proposal 
+    // Array.prototype.at(index): https://tc39.es/proposal-relative-indexing-method/#sec-array.prototype.at
+    // Spec: https://tc39.es/proposal-relative-indexing-method
+    // Github: https://github.com/tc39/proposal-relative-indexing-method
+    Var JavascriptArray::EntryAt(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+        
+        ARGUMENTS(args, callInfo);
+        ScriptContext* scriptContext = function->GetScriptContext();
+        JS_REENTRANCY_LOCK(jsReentLock, scriptContext->GetThreadContext());
+        AUTO_TAG_NATIVE_LIBRARY_ENTRY(function, callInfo, _u("Array.prototype.at"));
+
+        Assert(!(callInfo.Flags & CallFlags_New));
+
+        CHAKRATEL_LANGSTATS_INC_BUILTINCOUNT(Array_Prototype_at);
+
+        if (args.Info.Count == 0)
+        {
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NullOrUndefined, _u("Array.prototype.at"));
+        }
+
+        BigIndex length;
+        JavascriptArray* pArr = nullptr;
+        RecyclableObject* obj = nullptr;
+
+        JS_REENTRANT(jsReentLock, TryGetArrayAndLength(args[0], scriptContext, _u("Array.prototype.at"), &pArr, &obj, &length));
+
+        if (length.IsSmallIndex())
+        {
+            JS_REENTRANT_UNLOCK(jsReentLock, return JavascriptArray::AtHelper(pArr, nullptr, obj, length.GetSmallIndex(), args, scriptContext));
+        }
+        Assert(pArr == nullptr || length.IsUint32Max()); // if pArr is not null lets make sure length is safe to cast, which will only happen if length is a uint32max
+        JS_REENTRANT_UNLOCK(jsReentLock, return JavascriptArray::AtHelper(pArr, nullptr, obj, length.GetBigIndex(), args, scriptContext));
+    }
+
+    template <typename T>
```

Comment:
Why does this need to be a template method? (I think the CI failures are to do with failing to instantiate some forms of this template)

Reply:
I think, you guessed wrong - then EntryHelper also shouldn't to "instantiate some forms of this template".
https://github.com/chakra-core/ChakraCore/blob/master/lib/Runtime/Library/JavascriptArray.cpp#L8773-L8883
(Sorry, if this statment sounds sarcastic.)

Reply:
The macOS X build failed with:
Undefined symbols for architecture x86_64:
  "void* Js::JavascriptArray::AtHelper<unsigned int>(Js::JavascriptArray*, Js::TypedArrayBase*, Js::RecyclableObject*, unsigned int, Js::Arguments&, Js::ScriptContext*)", referenced from:
      Js::TypedArrayBase::EntryAt(Js::RecyclableObject*, Js::CallInfo, ...) in libChakraCoreStatic.a(TypedArray.cpp.o)

Perhaps need to make the template instantiation explicit.

Reply:
Explicit template instantiation means something like that?
```cpp
template <uint32_t>
 Var JavascriptArray::AtHelper(JavascriptArray* pArr, Js::TypedArrayBase* typedArrayBase, RecyclableObject* obj, uint32_t length, Arguments& args, ScriptContext* scriptContext);
```

Reply:
With the update you've done it builds on maOS now - though TBH I just get confused with template instantiation sometimes seeing as the existing SliceHelper was ok and set up the same way also - your original code would build with gcc but not with clang - I guess clang is just a bit too finicky sometimes.